client/my-sites/marketing/tools/index.tsx

```diff
@@ -0,0 +1,96 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import page from 'page';
+import PropTypes from 'prop-types';
+import React, { Fragment, FunctionComponent } from 'react';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import ActionCard from 'components/action-card';
+import Button from 'components/button';
+import { getSelectedSiteSlug } from 'state/ui/selectors';
+import MarketingToolFeature from './feature';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+interface MarketingToolsProps {
+	selectedSiteSlug: string | null;
```

Comment:
Shouldn't we use the `?` syntax for that?
```javascript
selectedSiteSlug?: string;
```

Reply:
In this case I think we will always be giving this prop via the connect below, so it will always be present in the interface, but one of those two values.

I actually tried `selectedSiteSlug?: string;` and VS Code threw a fit because it translates `?: type` into `type | undefined`

Reply:
Gotcha. I misunderstood the purpose of `?` - this is only to denote an [optional parameter](https://www.typescriptlang.org/docs/handbook/functions.html#optional-and-default-parameters).

Reply:
It's valid for [optional properties](https://www.typescriptlang.org/docs/handbook/interfaces.html) in an interface too.

We can think of `?` as specifying `undefined` vs. what we're doing here which is `null`